### PR TITLE
Use elixir_expand to simplify guard tests

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/elixir/test/elixir/module/types/infer_test.exs [hunk 1]
```diff
@@ -14,10 +14,18 @@ defmodule Module.Types.InferTest do
...
   defmacrop quoted_guard(guards, context) do
     quote do
-      of_guard(unquote(Macro.escape(guards)), new_stack(), unquote(context))
+      of_guard(unquote(Macro.escape(expand_guards(guards))), new_stack(), unquote(context))
     end
   end
 
+  defp expand_guards(guards) do
+    x = nil
+    _ = x
```
Comment (ericmj) [inline lib/elixir/test/elixir/module/types/infer_test.exs]: Is there a better to introduce variables to the environment? Even if I manually add them to `__ENV__.vars` I still get errors.

Comment (josevalim) [reply lib/elixir/test/elixir/module/types/infer_test.exs]: @ericmj I think the most reliable way would be for you to expand a whole expression, such as `fn x when unquote(guards) -> :ok end` and then you extract the guards out. WDTY?

**Comment by ericmj** [comment]
@josevalim Please check the last commit, is that what you had in mind?

**Comment by josevalim** [comment]
Exactly. Does it solve the issues you had in mind accordingly?
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Pull request merged**